target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-mr8300.dts

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+
+#include "qcom-ipq4019-xx8300.dtsi"
+
+/ {
+	model = "Linksys MR8300 (Dallas)";
+	compatible = "linksys,mr8300", "qcom,ipq4019";
+
+
+	aliases {
+		led-boot = &led_blue;
+		led-failsafe = &led_red;
+		led-running = &led_blue;
+		led-upgrade = &led_amber;
+		serial0 = &blsp1_uart1;
+	};
+
+		// Top panel LEDs, above Linksys logo
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_red: red {
+			label = "mr8300:red";
```

Comment:
Well, the labels-property has been deprecated in upstream.:
<https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/leds/common.yaml#L59>

So, you might as well drop it... Sadly we don't have the color and functions bindings in our trees yet.

Reply:
Do you already have an idea (or "vision") how OpenWrt should handle this in the future?

Reply:
Is there something to handle here in a device review? The upstream patch that changed the binding and deprecated "label" is: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=c5d18dd6b64e09dd6984bda9bdd55160af537a8c

   $ git describe c5d18dd6b64e09dd6984bda9bdd55160af537a8c
   v5.3-rc1-18-gc5d18dd6b64e

So I was wrong. 5.4 should have the color and function bindings :)

Reply:
So we are okay here?

Reply:
> So we are okay here?

@adschm what's your opinion there? I mean dropping the label property and adding the color property should be easy.
Not sure about the function of this particular LED though. @cybrnook can you look into the <dt-bindings/leds/common.h> and pick something that matches the function of the LED?

Reply:
@chunkeey I don't have enough time right now to build an opinion on how to do this best in the future. I was mainly looking for an entry point to do that at some point (when asking you about the future handling, as I've noticed your other commit beforehand).

Anyway, I won't merge device support on ipq40xx, so maybe just ignore me for now and do what seems best ...

Reply:
> Anyway, I won't merge device support on ipq40xx, so maybe just ignore me for now and do what seems best ... (or leave the decision to whoever will merge it)

Hmm? Since 9ba09653ade1fa20d0d777390d522fb179c1088e from @ynezz there are no target maintainers anymore. Anyone is welcome to merge.

As for why this came up. His labels for "mr8300:red", "mr8300:amber" and "mr8300:blue" are not matching the deprecated idea that the label encodes the "device:color:function". Looking at the EA8300 it seemed like some of these are for WPS, diagnose, etc.. So it might be a good idea to start with the slow conversion here and update the rest once the next stable kernel rolls around.

For supported values of the color and function property please look at 5.4's include/dt-bindings/leds/common.h

https://elixir.bootlin.com/linux/v5.4.66/source/include/dt-bindings/leds/common.h

Reply:
Would you like me to specify a function to the color vs just color for color? I think that would be easy enough.

Reply:
I have added labels for the LED colors and intended function based on the above link:
![image](https://user-images.githubusercontent.com/22136734/93658211-933a2d80-fa07-11ea-85e5-58a0e807db33.png)

Reply:
k, well, it's consistent with the other devices. I guess we'll do the remove the lables and introduce the color and function properties once we update to the next stable kernel.

Reply:
OT here, but I'm currently starting to play around with it. It didn't work out of the box though (generic status leds seem broken due to issues with the path, and I'm not satisfied with converting e.g. wlan2g and wlan5g to just green:wlan and green:wlan_1)

Reply:
Hmm, I will try to upstream LED_FUNCTION_WLAN2G, LED_FUNCTION_WLAN5G and LED_FUNCTION_RSSI ...

Reply:
@adschm hmm? problems with path? The MR32 was working fine, even though the PWM_LEDs are a bit finky. Aren't all the led-alias lookups funneled through the get_dt_led() script?

Reply:
@chunkeey Let's move the discussion here: https://github.com/openwrt/openwrt/pull/3437#issuecomment-695351722